docs/copy-backend.md

```diff
@@ -11,6 +11,8 @@ Components of the backend being copied:
 * methods
 * mapping rules
 
+If a backend with the selected `system-name` is found, it will be updated. Only missing metrics, methods and mapping rules will be created new.
```

Comment:
```suggestion
If a backend with the selected `system-name` is found, it will be updated. Only missing metrics, methods and mapping rules will be created.
```

Reply:
What happens with existing metrics, methods and mapping rules that have changed? will be updated?

Reply:
Backend metrics, methods and mapping rules are not updated. 

* Backend metrics are equal when ANY of the attributes`[system_name friendly_name]` is equal
* Backend methods are equal when ANY of the attributes`[system_name friendly_name]` is equal
* Backend mapping rules are euql when ALL of the attributes `[pattern http_method delta metric_id]` are equal

Reply:
Let me know if other policy should be used.

Reply:
"Backend metrics are equal when ANY of the attributes[system_name friendly_name] is equal"

so imagine i have 2 metrics in source product:

[system name, friendly name]
metric_1, one
metric_2, two

and in target product:

[system name, friendly name]
metric_2, one
metric_1, two

what happens?

Reply:
the command `3scale backend copy` does not create or update any metric. Kind of no-op for metrics for this scenario. 

However, `friendly_name` is also unique. So, if the toolbox would try to update target backend metric `metric_1` with a friendly_name `one`, then the operation would fail with the following error:
```
---
errors:
  friendly_name:
  - has already been taken
```

Another policy would be delete all target metrics first, and then copy. The toolbox would make sure target would be exactly as source backend. However, it would be destructive. Any metric created by the customer would be deleted.

Reply:
For copying mapping rules, the toolbox needs to build a map between source metric id's and target metric id's. For example, in this scenario,


```
source 
[system name, id ]
metric_1, 1000
metric_2, 2000

target 
[system name, id ]
metric_1, 9999
metric_2, 8888
```
the mapping would be:

```
{
1000 => 9999,
2000 => 8888
}
```

To build that map, it is assumed that `system_name` are equal both in source and target backends. Thus, I guess we need to change metric / method copy policy to:

```
Backend metrics are equal when system_name is equal
Backend methods are equal  when system_name is equal
```

Reply:
yes, i think the system name should be the boss. Also because it is not editable and friendly name is

Reply:
so imagine i have 2 metrics in source backend:

[system name, friendly name]
metric_1, one
metric_2, two

and in target backend:

[system name, friendly name]
metric_2, one
metric_1, two

what happens?

end state target:

[system name, friendly name]
metric_1, one
metric_2, two

Reply:
Finally: 
If a metric/method with the same `system_name` is found at the target backend, then that metric/method will not be created. And not updated (its friendly name)